handlers/restart_consul.yml

```
@@ -6,13 +6,14 @@
   when: ansible_service_mgr == "systemd"
   listen: reload systemd daemon
 
-- name: Restart consul on Unix
-  ansible.builtin.service:
-    name: consul
-    state: restarted
-    # Needed to force SysV service manager on Docker for Molecule tests
-    use: "{{ ansible_service_mgr }}"
+- name: Rolling restart consul on Unix
+  # Manual loop on playbook hosts to perform a "serial: 1" equivalent on a set of task
+  include_tasks: ../tasks/leave_restart_consul.yml
+  with_items: "{{ play_hosts }}"
+  loop_control:
+    loop_var: consul_servers
```

```suggestion
    loop_var: consul_server
```

Not to nitpick, but it would be nice if the `loop_var` was singular instead of plural, because it's a bit confusing then in `leave_restart_consul.yml` and looks like the tasks are being delegated to multiple hosts.

---

No worries, my bad, `rolling_restart_host` is better, let use it!